src/app.ts

```diff
@@ -112,6 +112,13 @@ if (process.env.NODE_ENV === 'production') {
     }
   )
   server.listen(443)
+
+  const httpApp = express()
+  httpApp.get('/*', (req, res) => {
+    const params = req.query ? `?${req.headers.host.split('?')[1]}` : ''
+    res.redirect(`https://${req.headers.host}${req.path}/${params}`)
```

Comment:
1. question: Do you need `/` between host and path, or does path already prefixed by `/`?
2. query perimeters: you need to add `?` right?